File: slack-reader/src/test/java/dev/number6/slackreader/dagger/TestSlackReaderComponent.java
package dev.number6.slackreader.dagger;

import dev.number6.db.dagger.DatabaseConfigurationPortModule;
import dev.number6.db.dagger.DatabasePortModule;
import dev.number6.slack.port.HttpPort;
import dev.number6.slackreader.port.SlackPort;
import dagger.Component;

import javax.inject.Singleton;
import java.time.Clock;

@Component(modules = {FakeHttpModule.class,
//        FakeAWSSecretsManagerModule.class,
        FakeDynamoDbMapperModule.class,
        DatabasePortModule.class,
        DatabaseConfigurationPortModule.class,
        SlackServiceModule.class,
        RecordingSlackPortModule.class,
        FakeAmazonSnsModule.class,
        FakeConfigurationPortModule.class,
        FakeClockModule.class})
@Singleton
public interface TestSlackReaderComponent extends SlackReaderComponent {

    HttpPort getFakeHttpAdaptor();

    FakeDynamoDbMapperModule.FakeAmazonDynamoDB getFakeAmazonDynamoClient();

    FakeAmazonSnsModule.FakeAmazonSns getFakeAmazonSns();

    Clock getClock();

    SlackPort getSlackPort();
}

File: slack-reader/src/test/kotlin/dev/number6/slackreader/dagger/TestSlackReaderComponent.kt
package dev.number6.slackreader.dagger

import dagger.Component
import dev.number6.db.dagger.DatabaseConfigurationPortModule
import dev.number6.db.dagger.DatabasePortModule
import dev.number6.slack.dagger.SlackClientModule
import dev.number6.slackreader.dagger.FakeAmazonSnsModule.FakeAmazonSns
import dev.number6.slackreader.port.SlackReaderPort
import java.time.Clock
import javax.inject.Singleton

@Component(modules = [FakeDynamoDbMapperModule::class,
    DatabasePortModule::class,
    DatabaseConfigurationPortModule::class,
    SlackServiceModule::class,
    RecordingSlackPortModule::class,
    SlackClientModule::class,
    FakeCallFactoryModule::class,
    FakeAmazonSnsModule::class,
    FakeConfigurationPortModule::class,
    FakeClockModule::class])
@Singleton
interface TestSlackReaderComponent : SlackReaderComponent {
    fun getFakeAmazonDynamoClient(): FakeDynamoDbMapperModule.FakeAmazonDynamoDB
    fun getFakeAmazonSns(): FakeAmazonSns
    fun getClock(): Clock
    fun getSlackPort(): SlackReaderPort
}